### PR TITLE
feat: self hosted agent pool

### PR DESCRIPTION
**Pull request by Dmytrovel**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -112,3 +112,35 @@ variable "variables_set" {
...
+
+variable "self_hosted_linux_agent_enable" {
+  type        = bool
+  description = "Self hosted linux agent enable"
+  default     = false
+}
+
+variable "pool_configuration" {
+  type = object({
+    agent_name     = string
```
Comment (owlleg6) [inline variables.tf:124]: rename parameter to "name"

File: variables.tf [hunk 2]
```diff
@@ -112,3 +112,35 @@ variable "variables_set" {
...
+    auto_provision = optional(bool)
+    auto_update    = optional(bool)
+  })
+  description = "Objects to configure pool"
+  default = {
```
Comment (owlleg6) [inline variables.tf:129]: provide default values, for example name is 'iacda', auto provision - true and auto update - true.

File: main.tf [hunk 3]
```diff
@@ -66,3 +66,28 @@ resource "azuredevops_variable_group" "this" {
...
+  agent_pool_id = azuredevops_agent_pool.this[0].id
+}
+
+resource "azuredevops_resource_authorization" "this" {
+  count = var.self_hosted_linux_agent_enable ? 1 : 0
+
+  project_id  = var.existing_project_name == null ? azuredevops_project.this[0].id : data.azuredevops_project.existing[0].id
+  resource_id = azuredevops_agent_queue.this[0].id
+  type        = var.pool_authorization.type
+  authorized  = var.pool_authorization.authorized
```
Comment (owlleg6) [inline main.tf:91]: add white space before 'depends on' block

File: variables.tf [hunk 4]
```diff
@@ -112,3 +112,35 @@ variable "variables_set" {
...
+  default     = false
+}
+
+variable "pool_configuration" {
+  type = object({
+    agent_name     = string
+    auto_provision = optional(bool)
+    auto_update    = optional(bool)
+  })
+  description = "Objects to configure pool"
```
Comment (owlleg6) [inline variables.tf:128]: Update description to - 'Object with parameters to configure self-hosted agent pool'

File: variables.tf [hunk 5]
```diff
@@ -112,3 +112,35 @@ variable "variables_set" {
...
+    auto_update    = false
+  }
+}
+
+variable "pool_authorization" {
+  type = object({
+    type       = string
+    authorized = bool
+  })
+  description = "Objects to configure pool authorization"
```
Comment (owlleg6) [inline variables.tf:141]: Update description to - 'Object with parameters to configure authorization of self-hosted agent pool'

File: outputs.tf [hunk 6]
```diff
@@ -0,0 +1,9 @@
+output "id" {
+  value       = var.self_hosted_linux_agent_enable ? azuredevops_agent_pool.this[0].id : ""
```
Comment (owlleg6) [inline outputs.tf:2]: you can use 'try' here, for example - try(azuredevops_agent_pool.this[0].id, null)

File: outputs.tf [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  description = "The ID of the agent pool."
+}
+
+output "name" {
+  value       = var.self_hosted_linux_agent_enable ? azuredevops_agent_pool.this[0].name : ""
```
Comment (owlleg6) [inline outputs.tf:7]: you can use 'try'

**Review by owlleg6 — CHANGES_REQUESTED** [review]
Review comments

File: outputs.tf [hunk 8]
```diff
@@ -0,0 +1,9 @@
+output "id" {
+  value       = try(azuredevops_agent_pool.this[0].id, null) # var.self_hosted_linux_agent_enable ? azuredevops_agent_pool.this[0].id : ""
```
[the remark this answers is not in the document]
Comment (Dmytrovel) [reply outputs.tf]: +

[the remark this answers is not in the document]
Comment (Dmytrovel) [reply outputs.tf]: done

**Review by owlleg6 — APPROVED** [review]

**Pull request merged**